book/src/out-of-order-blocks.md

```diff
@@ -0,0 +1,133 @@
+# Out Of Order Blocks
+
+The [deterministic leader schedule](../tree/master/book/src/leader-rotation.md)
+creates a fixed schedule for block production.  The main problems with this
+approach are as follows:
+
+* The leader schedule has to balance both performance and liveness requirements.
+The epoch needs to be long enough to guarantee liveness.
+
+* The cluster has to wait for unavailable nodes.
+
+* The cluster's liveness depends on the availability of scheduled nodes.
+
+* Attackers know in advance all of the leaders that propose a block.
+
+The general problems of DPOS — that a cluster requires an interactive protocol
+for account registration, delegation, and voting that a set of leaders can
+censor — is not addressed in this design.
+
+## Random Leaders
+
+Leaders can generate a *golden ticket* to propose a subsequent block.
+
+* *N* number of bits of a leader's vote signature that must match a PoH hash.
+This hash can be in between ticks.
+
+The leader must find the golden ticket in the empty PoH entries leading up to
+the proposed block.  Once the golden ticket is found, the leader can propose the
+block for the next slot.
+
+To find the golden ticket, the leader signs the BlockHash of the block that the
```

Comment:
`the leader signs`, this can be any validator searching for the golden ticket for block `N+1`? It doesn't have to be the current leader for block `N`?

Reply:
@carllin correct, thanks!